src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -193,16 +228,31 @@ public override SdkResult Resolve(SdkReference sdkReference, SdkResolverContext
             {
                 return Failure(
                     factory,
+                    logger,
+                    context.Logger,
                     Strings.MSBuildSDKDirectoryNotFound,
                     msbuildSdkDir);
             }
 
             return factory.IndicateSuccess(msbuildSdkDir, netcoreSdkVersion, propertiesToAdd, itemsToAdd, warnings);
         }
 
-        private static SdkResult Failure(SdkResultFactory factory, string format, params object[] args)
+        private static SdkResult Failure(SdkResultFactory factory, ResolverLogger logger, SdkLogger sdkLogger, string format, params object[] args)
         {
-            return factory.IndicateFailure(new[] { string.Format(format, args) });
+            string error = string.Format(format, args);
+            logger.LogMessage($"Failed to resolve SDK: {error}");
```

Comment:
Yes, but I thought it was best to make sure all the information was included, rather than potentially leaving some out, if there's a case where you end up just with the log but not the error.  Maybe that's not possible depending on how MSBuild handles this data on its side.

Reply:
If SDK resolution fails entirely, MSBuild will log all warnings and errors from all SDK resolvers.  If one resolver fails and the next one succeeds, then only messages and warnings are logged.  

https://github.com/dotnet/msbuild/blob/a2490dd3f78cce4abc8f9e6f1b5268437332818f/src/Build/BackEnd/Components/SdkResolution/SdkResolverService.cs#L302

So yeah I guess it might be desirable to see any errors just in case another SDK resolver succeeds but really shouldn't be handling the call.

Reply:
But do we know for sure at this point that it really should be being handled by the MSBuildSdkResolver? MSBuild naively tries any resolver that looks like it _might_ be able to resolve the SDK, but that doesn't mean this is right, and I don't think we should error if another SDK resolver can figure it out unless we're sure they're wrong.